src/clib/pio_getput_int.c

```diff
@@ -1042,17 +1042,23 @@ int PIOc_get_vars_tc(int ncid, int varid, const PIO_Offset *start, const PIO_Off
             else
             {
                 /* non-root reads nothing */
-                cnt = (PIO_Offset*) calloc(ndims, sizeof(PIO_Offset));
-                if (cnt == NULL)
+                if (ndims == 0)
+                    cnt = NULL; /* scalar variables can pass NULL for start/count */
+                else
                 {
-                    GPTLstop("PIO:PIOc_get_vars_tc");
-                    spio_ltimer_stop(ios->io_fstats->rd_timer_name);
-                    spio_ltimer_stop(ios->io_fstats->tot_timer_name);
-                    spio_ltimer_stop(file->io_fstats->rd_timer_name);
-                    spio_ltimer_stop(file->io_fstats->tot_timer_name);
-                    return pio_err(ios, file, PIO_ENOMEM, __FILE__, __LINE__,
-                                   "Reading variable (%s, varid=%d) from file (%s, ncid=%d) failed. Out of memory allocating %lld bytes for a temporary count array",
-                                   pio_get_vname_from_file(file, varid), varid, pio_get_fname_from_file(file), ncid, (long long int) (ndims * sizeof(PIO_Offset)));
+                    assert(ndims > 0);
```

Comment:
if you want to check for ndims < 0 maybe it should be checked where ndims is initially set.

Reply:
OK, I have updated the check for ndims.